python/lsst/daf/butler/core/time_utils.py

```diff
@@ -37,135 +37,154 @@
 except ImportError:
     erfa = None
 
-# These constants can be used by client code.
-# EPOCH is used to construct times as read from database, its precision is
-# used by all those timestamps, set it to 1 microsecond.
-EPOCH = astropy.time.Time("1970-01-01 00:00:00", format="iso", scale="tai", precision=6)
-"""Epoch for calculating time delta, this is the minimum time that can be
-stored in the database.
-"""
-
-MAX_TIME = astropy.time.Time("2100-01-01 00:00:00", format="iso", scale="tai")
-"""Maximum time value that we can store. Assuming 64-bit integer field we
-can actually store higher values but we intentionally limit it to arbitrary
-but reasonably high value. Note that this value will be stored in registry
-database for eternity, so it should not be changed without proper
-consideration.
-"""
-
-# number of nanosecons in a day
-_NSEC_PER_DAY = 1_000_000_000 * 24 * 3600
+from .utils import Singleton
 
 _LOG = logging.getLogger(__name__)
 
 
-def astropy_to_nsec(astropy_time: astropy.time.Time) -> int:
-    """Convert astropy time to nanoseconds since epoch.
-
-    Input time is converted to TAI scale before conversion to
+class TimeConverter(metaclass=Singleton):
+    """A singleton that provides methods for mapping TAI times to integer
```

Comment:
A bit surprising to see singleton used to solve this problem, but probably fine. I guess I just don't like singletons in general 🙂.